src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -391,19 +385,16 @@ FixedwingPositionControl::adapt_airspeed_setpoint(const float control_interval,
 	calibrated_airspeed_setpoint = constrain(calibrated_airspeed_setpoint, calibrated_min_airspeed,
 				       _performance_model.getMaximumCalibratedAirspeed());
 
-	if (!PX4_ISFINITE(_airspeed_slew_rate_controller.getState())) {
-		_airspeed_slew_rate_controller.setForcedValue(calibrated_airspeed_setpoint);
+	if (!PX4_ISFINITE(_airspeed_slew_rate_controller.getState())
+	    || _airspeed_slew_rate_controller.getState() < calibrated_min_airspeed || !_tecs_is_running) {
+		_airspeed_slew_rate_controller.setForcedValue(math::max(calibrated_min_airspeed, _airspeed_eas));
```

Comment:
if _airspeed_slew_rate_controller.getState() < calibrated_min_airspeed you want to set it always to calibrated_min_airspeed not to _airspeed_eas. For the other two conditions, it is fine.

Reply:
Good catch, done in